packages/headless/docs.config.json

```diff
@@ -0,0 +1,279 @@
+{
+  "engine": {
+    "sources": [
+      "src/app/headless-engine"
+    ]
+  },
+  "actions": {
+    "sections": [
+      {
+        "name": "Configuration",
+        "sources": [
+          "src/features/configuration/configuration-actions"
+        ]
+      },
+      {
+        "name": "Did-You-Mean",
+        "sources": [
+          "src/features/did-you-mean/did-you-mean-actions"
+        ]
+      }
+    ]
+  },
+  "controllers": [
+    {
+      "name": "BreadcrumbManager",
+      "initialize": {
+        "initializer": {
+          "name": "buildBreadcrumbManager",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        "others": []
+      },
+      "state": "BreadcrumbManagerState",
+      "related": [
+        {
+          "name": "BreadcrumbValue",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        {
+          "name": "FacetBreadcrumb",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        {
+          "name": "NumericFacetBreadcrumb",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        {
+          "name": "DateFacetBreadcrumb",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        {
+          "name": "Breadcrumb",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        },
+        {
+          "name": "CategoryFacetBreadcrumb",
+          "source": "src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts"
+        }
+      ]
+    },
+    {
+      "name": "Facet",
+      "initialize": {
+        "initializer": {
+          "name": "buildFacet",
+          "source": "src/controllers/facets/facet/headless-facet.ts"
+        },
+        "others": [
+          {
+            "name": "FacetProps",
+            "source": "src/controllers/facets/facet/headless-facet.ts"
+          },
+          {
+            "name": "FacetOptions",
+            "source": "src/controllers/facets/facet/headless-facet-options.ts"
+          }
+        ]
+      },
+      "state": "FacetState",
+      "related": [],
+      "subcontrollers": [
```

Comment:
In order to extract any of the FacetSearch-related things that we _do_ want documented, i.e. state and methods, we need to extract `buildFacetSearch`, because both the `state` accessor and the various methods appear in the doc.json as part of the `buildFacetSearch` entity, not separately.
It is true, however, that we probably don't need _all_ the same information as for a controller. Instead of making `subcontrollers` be type `Controller[]`, I can make a new type `Subcontroller` that only has exactly what we need.

Reply:
I think reusing the same type is ok, as long as we do not document the initializer. The initialization is handled by the Facet controller internally. The `FacetSearchProps` and `FacetSearchOptions` are not directly exported.

Any facet search options are exposed via `FacetOptions`, so if the latter interface is documented, then so are the facet search options by extension. Could that simplify things?

Reply:
Resolving options (for any controller, but I'll focus on facet and facetsearch) is equivalent to resolving types from the doc.json file. right now, even regular `FacetOptions` don't really work properly. Because `FacetOptions` draws on other types and interfaces to construct itself, that's exactly how the type is listed in the doc.json. We don't actually have facet options at the moment, we just have some _Omit_ and _Partial_ and _Pick_ with various other types and interfaces. Try running the script (DocJsonParser.ts, soon to be doc-json-parser.ts) and looking at the Facet section of the generated parsed_doc.json.

Reply:
I see , it sounds like we're going to have to write code to resolve the types, hmm.

Reply:
I mean, that's already being done to some extent in TypeResolver.ts; that's how we get the types as well as we do right now. I don't know if we'll have to hardcode something separate for facets or what the answer is...